### PR TITLE
Improvement on the policy recommendation status command

### PR DESCRIPTION
**Pull request by dreamtalen**
In this commit, we are adding the following improvements on the policy
recommendation status command:
1. Check the ClickHouse DB for the completed jobs which are no longer on
   the API server.
2. Show the error message for the failed jobs.
3. Replace the empty state string with NEW for the jobs on NewState

### CODE REVIEW

File: pkg/theia/commands/utils.go [hunk 1]
```diff
@@ -156,3 +160,88 @@ func ResolveKubeConfig(cmd *cobra.Command) (string, error) {
...
+	connRetryInterval := 1 * time.Second
+	connTimeout := 10 * time.Second
+
+	// Connect to ClickHouse in a loop
+	if err := wait.PollImmediate(connRetryInterval, connTimeout, func() (bool, error) {
+		// Open the database and ping it
+		var err error
+		connect, err = sql.Open("clickhouse", url)
+		if err != nil {
+			connErr = fmt.Errorf("failed to ping ClickHouse: %v", err)
```
Comment (yuntanghsu) [inline pkg/theia/commands/utils.go:192]:
```suggestion
			connErr = fmt.Errorf("failed to open ClickHouse: %v", err)
```

**Comment by yanjunz97** [comment]
Maybe also remove `use-cluster-ip` and `clickhouse-endpoint` flags from `policy_recommendation_run.go`?

File: pkg/theia/commands/policy_recommendation_status.go [hunk 2]
```diff
@@ -67,73 +68,118 @@ $ theia policy-recommendation status e998433e-accb-4888-9fc8-06563f073e86 --use-
...
+	err = clientset.CoreV1().RESTClient().
 		Get().
 		AbsPath("/apis/sparkoperator.k8s.io/v1beta2").
 		Namespace(flowVisibilityNS).
 		Resource("sparkapplications").
-		Name("pr-" + recoID).
+		Name("pr-" + id).
 		Do(context.TODO()).
-		Into(sparkApplication)
+		Into(&sparkApp)
```
Comment (heanlan) [inline pkg/theia/commands/policy_recommendation_status.go:155]: Apology if I missed something... Where does the `sparkApp` come from? I didn't see its initialization.

Comment (dreamtalen) [reply pkg/theia/commands/policy_recommendation_status.go:155]: In the return value list of this function.

**Review by yanjunz97 — APPROVED** [review]
LGTM

**Review by heanlan — APPROVED** [review]
LGTM

**Pull request merged**